src/main/java/com/github/fge/jsonpatch/JsonPathParser.java

```diff
@@ -1,52 +1,17 @@
 package com.github.fge.jsonpatch;
 
-import static com.github.fge.jsonpatch.JsonPatchOperation.BUNDLE;
-
 public class JsonPathParser {
 
-    private static final String ARRAY_ELEMENT_REGEX = "\\.(\\d+)\\.";
-    private static final String ARRAY_ELEMENT_LAST_REGEX = "\\.(\\d+)$";
+    private static final String ARRAY_ELEMENT_REGEX = "(?<=\\.)(\\d+)";
 
     public static String tmfStringToJsonPath(String path) throws JsonPatchException {
-        if (path.startsWith("$") && !path.isEmpty()) {
+        if (path.startsWith("$")) {
             return path;
         }
-        if (!path.startsWith("/") && !path.isEmpty()) {
-            return "$." + path;
-        }
-        if ("/".equals(path)) {
-            return "$";
-        }
-        final String[] pointerAndQuery = path
-                .replaceAll("(\\w)\\?", "$1#THIS_IS_SPLIT_PLACEHOLDER#")
-                .split("#THIS_IS_SPLIT_PLACEHOLDER#", -1);
-        if (pointerAndQuery.length > 2) {
-            throw new JsonPatchException(BUNDLE.getMessage("jsonPatch.invalidPathExpression"));
+        if (path.contains("?")) {
```

Comment:
Looks like a limitation that should be documented.

Question mark can appear in:
* field name - query: `$.['alive?']`
* filter - `[?(@.name empty true)]`

Reply:
These queries would be returned without modifications in line 8
```
if (path.startsWith("$")) {
            return path;
        }
```